src/Message/PurchaseRequest.php

```diff
@@ -112,4 +116,160 @@ private function resolvePageAction()
 
         return 'PAYMENT';
     }
+
+    public function getCaptureDelay()
+    {
+        $captureDelay = $this->getParameter('captureDelay');
+        if (null !== $captureDelay) {
+            return $captureDelay;
+        }
+
+        return 0;
+    }
+
+    public function setCaptureDelay($value)
+    {
+        return $this->setParameter('captureDelay', $value);
+    }
+
+    public function getPaymentConfig()
+    {
+        $paymentConfig = $this->getParameter('paymentConfig');
+        if (null !== $paymentConfig) {
+            return $paymentConfig;
+        }
+
+        return 'SINGLE';
+    }
+
+    /**
+     * @param string $type
+     * @param array $values
+     * @return PurchaseRequest
+     * @throws InvalidRequestException
+     */
+    public function setPaymentConfig($type, $values = [])
+    {
+        if ($type === 'SINGLE') {
+            return $this->setParameter('paymentConfig', $type);
+        }
+
+        if (!in_array($type, ['MULTI', 'MULTI_EXT'])) {
+            throw new InvalidRequestException('type is not allowed');
+        }
+
+        if (0 !== $this->getCaptureDelay()) {
```

Comment:
are we sure we want to break when the capture delay is set to something else than 0 ? 

In the documentation, it's written that the field will be ignored when MULTI_EXT is used (probably because date of transaction are already defined), but in case of SINGLE or MULTI we could have a delay before each transaction.